### PR TITLE
Add Waveshare RP2040 One to Circuit Python

### PR DESCRIPTION
**Pull request by johnnohj**
Submitting to include the Waveshare board [RP2040 One](https://www.waveshare.com/product/rp2040-one.htm) with Circuit Python. It's essentially an RP2040 Zero clone from the same company, except for the larger flash and the Trinkey-like form factor. 

In fact, it's so much of an RP2040 Zero clone that Waveshare doesn't offer a separate bootloader for it, which makes the Product ID something of an issue.  When I compiled a .u2if for it, I had temporarily assigned `0x111F` for the One , against `0x101F` for the Zero.  Such is my sense of humor.  

For this PR, I've adopted Waveshare's own practice and applied the Zero's Product ID as listed in the Circuit Python configuration files for that board.  Nevertheless, PR submitted as a draft pending guidance.

### CODE REVIEW

**Comment by dhalbert** [comment]
> In fact, it's so much of an RP2040 Zero clone that Waveshare doesn't offer a separate bootloader for it,

The RP2040 bootloader is in ROM, and identifies the same for every RP2040 on any board. So you can't distinguish based on that. To get an official PID for this board, see https://github.com/raspberrypi/usb-pid (explained in https://learn.adafruit.com/how-to-add-a-new-board-to-circuitpython/frequently-asked-questions#faq-3130481)

**Comment by johnnohj** [comment]
I misspoke; I meant to say that Waveshare doesn't offer a separate .uf2 for the One and instead links the file for the Zero. 

Being unfamiliar with the process, are there potential issues applying for a product id as someone unaffiliated with Waveshare?

**Comment by dhalbert** [comment]
Thanks for the clarification. The README in https://github.com/raspberrypi/usb-pid links to a Google Form. I think you should get Waveshare to ask for a new PID via that form. This issue, https://github.com/raspberrypi/usb-pid/pull/14, was submitted by @EngineerWill, who appears to work for Waveshare, based on their profile. You could contact them and ask them to request a new PID. Maybe the ping of them here will also be helpful.

**Comment by johnnohj** [comment]
Your prompting has led me to discover that Waveshare has registered `0x103A` as the product id for the One. I will update the files accordingly. Many thanks, Dan!

**Comment by johnnohj** [comment]
The Build CI / ports (raspberrypi) / board [...] test seems to be tripping over the entry for the flash chip. The product spec on the website lists it as "W25Q32JVSSIQ - 4MB NOR-Flash," which is twice the size of the Zero and it is by that number I refer to it in the .mk file. Again my inexperience is showing, but I knew I wished to avoid the controller being able to 'see' only 2MB.

File: ports/raspberrypi/boards/waveshare_rp2040_one/mpconfigboard.mk [hunk 1]
```diff
@@ -0,0 +1,11 @@
+USB_VID = 0x2E8A
+USB_PID = 0x103A
+USB_PRODUCT = "RP2040-One"
+USB_MANUFACTURER = "Waveshare Electronics"
+
+CHIP_VARIANT = RP2040
+CHIP_FAMILY = rp2
+
+EXTERNAL_FLASH_DEVICES = "W25Q32JVSSIQ"
```
Comment (dhalbert) [inline ports/raspberrypi/boards/waveshare_rp2040_one/mpconfigboard.mk:9]: Use this name for the flash chip. See https://github.com/adafruit/nvm.toml/blob/main/flash/winbond/W25Q32JVxQ.toml
```suggestion
EXTERNAL_FLASH_DEVICES = "W25Q32JVxQ"
```

**Review by dhalbert — CHANGES_REQUESTED** [review]

File: ports/raspberrypi/boards/waveshare_rp2040_one/mpconfigboard.mk (hunk 1 above)
Comment (johnnohj) [reply ports/raspberrypi/boards/waveshare_rp2040_one/mpconfigboard.mk]: All checks now pass

**Pull request merged**